test/integration/scheduler/queue_test.go

```diff
@@ -849,27 +849,30 @@ func TestCoreResourceEnqueue(t *testing.T) {
 					}
 				}
 
-				// Wait for the tt.pods to be present in the scheduling queue.
+				// Wait for the tt.pods to be present in the scheduling active queue.
 				if err := wait.PollUntilContextTimeout(ctx, time.Millisecond*200, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
-					pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
-					return len(pendingPods) == len(tt.pods), nil
+					return len(testCtx.Scheduler.SchedulingQueue.PodsInActiveQ()) == len(tt.pods), nil
```

Comment:
```suggestion
					return len(testCtx.Scheduler.SchedulingQueue.PodsInActiveQ()) == len(tt.pods) &&  len(pendingPods) == len(tt.pods), nil
```

Let's keep the both check, in case there's some error data in the `unschedulablePods`.

Reply:
Done